File: 정수내림차순으로배치하기/javascript/20221019/solution.js
// export const toIntArray = (number) => {
//   return number
//     .toString()
//     .split('')
//     .map((string) => +string);
// };

// export const toReversedOrder = (numbers) => {
//   return numbers.sort((a, b) => b - a);
// };

// export const joining = (numbers) => {
//   return Number(numbers.reduce((a, b) => a + b, ''));
// };

const solution = (number) => {
  return (
    (number + '')
      .split('')
      .sort((a, b) => b - a)
      .join('') * 1
  );
};

export default solution;

File: 정수내림차순으로배치하기/java/20221019/Solution.java
import java.util.Arrays;
import java.util.Comparator;
import java.util.stream.Collectors;

public class Solution {
    public long solution(long n) {
        return toLong(sort(n));
    }
    public String sort(long i) {
        return Arrays.stream(Long.toString(i).split("")).sorted(Comparator.reverseOrder()).collect(Collectors.joining());
    }

    public long toLong(String s) {
        return Long.parseLong(s);
    }
}


